### PR TITLE
Clear conflict for diagnosis

### PR DESCRIPTION
**Pull request by mithun-grmtech**
Please check asap

### CODE REVIEW

**Comment by mithun-grmtech** [comment]
Create a new pull request [click here](https://github.com/savantcare/emr/pull/95)

**Comment by mithun-grmtech** [comment]
It's a new request

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/client-side/structure/master-table-of-diagnosis.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  static primaryKey = 'masterDiagnosisId'
+
+  static fields() {
+    return {
+      ...super.fields(),
+
+      masterDiagnosisId: this.uid(() => intUniqueID()),
+      diagnosisName: this.string(null),
+
+      parity: this.number(0),
```
Comment (rajgrmtech) [inline webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/client-side/structure/master-table-of-diagnosis.js:23]: Line no: 23 -> 'parity' is not meaningful

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/client-side/structure/patient-table-of-diagnosis.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  static entity = 'tablePatientOfDiagnosis'
+
+  static primaryKey = 'clientSideUniqRowId'
+
+  static fields() {
+    return {
+      ...super.fields(),
+
+      clientSideUniqRowId: this.uid(() => intUniqueID()),
+      masterDiagnosisId: this.uid(() => uuidv1()), // This is service statement ID assigned to this patient coming from master table
```
Comment (rajgrmtech) [inline webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/client-side/structure/patient-table-of-diagnosis.js:20]: Value of 'masterDiagnosisId' should be 'this.number(null)'

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/server-side/structure-gen-on-2020-09-15.sql [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+--
+
+-- --------------------------------------------------------
+
+--
+-- Table structure for table `assignedDiagnosis`
+--
+
+CREATE TABLE `assignedDiagnosis` (
+  `uuid` char(36) NOT NULL,
```
Comment (rajgrmtech) [inline webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/server-side/structure-gen-on-2020-09-15.sql:13]: 'uuid' should be 'serverSideRowUuid'

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/server-side/structure-gen-on-2020-09-15.sql [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  `patientUUID` char(36) NOT NULL,
+  `masterDiagnosisId` int(11) NOT NULL,
+  `discontinueNote` varchar(255) DEFAULT NULL,
+  `discontinueOn` datetime(3) DEFAULT NULL,
+  `insertedOn` datetime(3) NOT NULL DEFAULT current_timestamp(3)
```
Comment (rajgrmtech) [inline webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/server-side/structure-gen-on-2020-09-15.sql:18]: Remove 'discontinueOn', 'insertedOn'
Add following fields:
`recordChangedByUUID` char(36) COLLATE utf8_unicode_ci DEFAULT NULL,
`recordChangedFromIPAddress` varchar(20) COLLATE utf8_unicode_ci DEFAULT NULL,

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/edit-layer/edit-design-1.vue [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+            )
+        })
+        .orderBy('parity', 'desc')
+        .get();
+        return arOfObjectsFromClientSideMasterDB;
+    }
+  },
+  mounted() {},
+  methods: {
+    fnSetMasterDataParity() {
```
Comment (rajgrmtech) [inline webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/edit-layer/edit-design-1.vue:53]: Useless function

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/view-layer/grid-with-cards-ct.vue [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  </div>
+</template>
+
+<script>
+
+import clientSideTblPatientDiagnosis from '../db/client-side/structure/patient-table-of-diagnosis'
+
+export default {
+  computed: { 
+    bindDisplayData: function() {
```
Comment (rajgrmtech) [inline webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/view-layer/grid-with-cards-ct.vue:60]: function name should be better.
This should be 'cfArOfDiagnosisForDisplay'

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/view-layer/grid-with-cards-ct.vue [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+      const getData = clientSideTblPatientDiagnosis
+        .query()
+        .with('linkWithMaster')
+        .where('ROW_END', 2147483647.999999)
+        .get();
+      return getData;
+    },
+   },
+  methods: {
+    fnDiscontinue(id) {
```
Comment (rajgrmtech) [inline webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/view-layer/grid-with-cards-ct.vue:70]: function name should be 'mfIconDeleteClickedOnChildCard'

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/client-side/structure/master-table-of-diagnosis.js (hunk 1 above)
Comment (mithun-grmtech) [reply webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/client-side/structure/master-table-of-diagnosis.js]: fixed

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/client-side/structure/patient-table-of-diagnosis.js (hunk 2 above)
Comment (mithun-grmtech) [reply webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/client-side/structure/patient-table-of-diagnosis.js]: changed done

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/server-side/structure-gen-on-2020-09-15.sql (hunk 3 above)
Comment (mithun-grmtech) [reply webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/server-side/structure-gen-on-2020-09-15.sql:13]: rename done

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/server-side/structure-gen-on-2020-09-15.sql (hunk 4 above)
Comment (mithun-grmtech) [reply webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/db/server-side/structure-gen-on-2020-09-15.sql:18]: all fixed

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/edit-layer/edit-design-1.vue (hunk 5 above)
Comment (mithun-grmtech) [reply webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/edit-layer/edit-design-1.vue]: Now the functions used

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/view-layer/grid-with-cards-ct.vue (hunk 6 above)
Comment (mithun-grmtech) [reply webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/view-layer/grid-with-cards-ct.vue]: rename

File: webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/view-layer/grid-with-cards-ct.vue (hunk 7 above)
Comment (mithun-grmtech) [reply webclient/components/pt-info/single/1time-Mrow-mField/diagnosis/view-layer/grid-with-cards-ct.vue]: rename done

**Review by rajgrmtech — APPROVED** [review]

**Review by vikaskedia — APPROVED** [review]

**Pull request merged**